Added documentations as README (#9)
GoogleCloudPlatform/opentelemetry-operations-collector: README.md

# OpenTelemetry Operations Collector

> This repository is focused on building and packaging the OpenTelemetry Collector with the necessary tools in different OSs to be compatible with Google Cloud Monitoring.

Reviewer: ```suggestion
This repository is focused on building and packaging the OpenTelemetry Collector for use with Google Cloud Monitoring.
```

## Prerequisite

[docker](https://docs.docker.com/engine/install/) is installed

## Building the Docker Image

Run `make docker-build-image`

> ## Collector for Linux

Reviewer: Using the docker image to run these make commands is useful if you want to replicate what the CI pipeline is doing directly, but most of the time users would just want to run the make commands directly imo. I'd probably have the initial example here call `make build-tarball`

Then add a brief paragraph at the end saying "alternatively you can run the make commands inside docker. First, generate a docker image for running the build commands by running `make docker-build-image` and then run `make TARGET=build-tarball docker-run`"

# OpenTelemetry Operations Collector

This repository is focused on building and packaging the OpenTelemetry Collector for use with Google Cloud Monitoring.

> ## Prerequisite
> 
> [docker](https://docs.docker.com/engine/install/) is installed
> 
> ## Building the Docker Image
> 
> Run `make docker-build-image`

Reviewer: ```suggestion
```

Author: Do you think we don't need these here?

Reviewer: I moved this section to the bottom in that other comment. You could leave the prerequisite section if you like.

## Collector for Linux

To generate a tarball file that packages the executable OpenTelemetry binary with the configuration file compatible with Google Cloud Monitoring.
1. Run `make build-tarball`
2. The tarball file will be generated in the `dist` folder

> Alternatively, you can  run the make commands inside docker.
> 1. Make sure docker image has already been built: run `docker images` to check if `otelopscol-build` image exists
> 2. If image doesn't exist, run `make docker-build-image` to build an image from the dockerfile
> 2. Run `make TARGET=build-tarball docker-run`
> 3. The tarball file will be generated in the `dist` folder

Reviewer: ```suggestion
## Running build commands in Docker

Alternatively, you can run the build commands inside docker:
1. Run `make docker-build-image` to build the docker image. This will generate an image called `otelopscol-build`
2. Run `make TARGET=build-<package> docker-run`
3. The specified package will be generated in the `dist` folder
```
